spot-client/src/common/remote-control/P2PSignalingBase.js

```diff
@@ -0,0 +1,353 @@
+     * the messages has been received.
+     * @param {Object} data - A message data with an ICE candidate.
+     * @private
+     * @returns {void}
+     */
+    _processIceCandidate(peerConnection, data) {
+        const candidate = this._parseJSONorLogError(data.candidate, 'ICE candidate');
+
+        if (!candidate) {
+            return;
+        }
+
+        peerConnection.addRemoteIceCandidate(candidate).catch(error => {
+            logger.error('Failed to set remote candidate', {
+                data,
+                error,
+                remoteAddress: peerConnection.remoteAddress
+            });
+        });
+    }
+
+    /**
+     * This method is to be called whenever P2P channel message arrives from the remote side. Those messages are
+     * emitted by the {@link P2PSignalingCallbacks.onSendP2PMessage} callback on the remote side and are to be set over
+     * the main signaling channel and set here.
+     *
+     * @param {Object} data - Message data.
+     * @param {string} from - Remote address used to match connections.
+     * @returns {void}
+     */
+    processP2PMessage({ data, from }) {
+        logger.log('processMessage', {
+            data,
+            from
+        });
+
+        if (data.offer) {
+            this._processOffer(from, data);
+
+            return;
+        }
+
+        const peerConnection = this.getConnectionForAddress(from);
+
+        if (!peerConnection) {
+            logger.error('processMessage - no PeerConnection', {
+                data,
+                from
+            });
+
+            return;
+        }
+
+        if (peerConnection.remoteAddress !== from) {
+            logger.error('Received message from different remote address', {
+                data,
+                from,
+                remoteAddress: peerConnection.remoteAddress
+            });
+
+            return;
+        }
+
+        if (data.answer) {
+            this._processAnswer(peerConnection, data);
+
+            return;
+        }
+
+        if (data.candidate) {
+            this._processIceCandidate(peerConnection, data);
+
+            return;
+        }
+
+        logger.warn('Skipped message processing', {
+            from,
+            data
+        });
+    }
+
+    /**
+     * Processing for offer message.
+     *
+     * @param {string} from - Remote address string.
+     * @param {string} data - Message data with an offer.
+     * @private
+     * @returns {void}
+     */
+    _processOffer(from, data) {
+        if (this.getConnectionForAddress(from)) {
+            logger.error('Received offer, but PeerConnection is still there', {
+                data,
+                from
+            });
+
+            return;
+        }
+
+        const offer = this._parseJSONorLogError(data.offer, 'offer message');
+
+        if (!offer) {
+            return;
+        }
+
+        // Try catch block in case something's wrong with WebRTC version or not supported
```

Comment:
When I was testing webview it was throwing errors because of incompatible API and I found that catching the error here prevents from breaking the app and makes P2P optional. After fix currently it's working everywhere where I tested it, but that does not cover older browsers.